src/mlx_omni_server/chat/mlx/exo_chat_generator.py

```
@@ -0,0 +1,304 @@
+            # Yield accumulated OpenAI tool calls as XML text so the
+            # Anthropic adapter's Qwen3MoeToolParser can process them
+            valid_tcs = [tc for tc in accumulated_tool_calls if tc["name"]]
+            for i, tc in enumerate(valid_tcs):
+                try:
+                    args = json.loads(tc["arguments"]) if tc["arguments"] else {}
+                except Exception:
+                    args = {}
+                xml_parts = [f"<tool_call>\n<function={tc['name']}>\n"]
+                for k, v in args.items():
+                    xml_parts.append(f"<parameter={k}>\n{v}\n</parameter>\n")
+                xml_parts.append(f"</function>\n</tool_call>")
```

Ruff (0.15.14)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)